packages/contracts-bedrock/contracts/L1/winddown/BalanceClaimer.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Libraries
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+// Interfaces
+import { IEthBalanceWithdrawer } from "../interfaces/winddown/IEthBalanceWithdrawer.sol";
+import { IErc20BalanceWithdrawer } from "../interfaces/winddown/IErc20BalanceWithdrawer.sol";
+import { IBalanceClaimer } from "../interfaces/winddown/IBalanceClaimer.sol";
+import { Semver } from "../../universal/Semver.sol";
+
+/**
+  * @custom:proxied
+  * @notice Contract that allows users to claim and withdraw their eth and erc20 balances
+ */
+contract BalanceClaimer is Initializable, Semver, IBalanceClaimer {
+    /// @notice the root of the merkle tree
+    bytes32 public root;
+
+    /// @notice OptimismPortal proxy address
+    IEthBalanceWithdrawer public ethBalanceWithdrawer;
+
+    /// @notice L1StandardBridge proxy address
+    IErc20BalanceWithdrawer public erc20BalanceWithdrawer;
+
+    /// @notice The mapping of users who have claimed their balances
+    mapping(address => bool) public claimed;
+
+    /**
+     * @custom:semver 1.7.0
+     */
+    constructor() Semver(1, 0, 0) {
+        initialize({_ethBalanceWithdrawer: address(0), _erc20BalanceWithdrawer: address(0), _root: bytes32(0)});
```

Comment:
what's the reason for  doing this instead of disabling the initializers?

Reply:
this is done in op contracts https://github.com/ethereum-optimism/optimism/blob/be455ca4fc0862d15f29f9964d53d54de0b9e4fa/packages/contracts-bedrock/src/L1/L1StandardBridge.sol#L89 constructor, initialize is called to set the proxy store values for the first time, if we make the 3 vars immutable we could just skip initialize